packages/amplify-graphql-api-construct-tests/src/__tests__/backends/all-auth-modes/app.ts

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+import 'source-map-support/register';
+import { App, Stack, Duration } from 'aws-cdk-lib';
+import { Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { UserPool, CfnIdentityPool } from 'aws-cdk-lib/aws-cognito';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+// @ts-ignore
+import { AmplifyGraphqlApi, AmplifyGraphqlDefinition } from '@aws-amplify/graphql-api-construct';
+import * as path from 'path';
+
+const packageJson = require('../package.json');
+
+const app = new App();
+const stack = new Stack(app, packageJson.name.replace(/_/g, '-'), {
+  env: { region: process.env.CLI_REGION || 'us-west-2' },
+});
+
+const userPool = new UserPool(stack, 'TestPool', {});
+// const identityPoolId = new CfnIdentityPool(stack, 'TestIdentityPool', { allowUnauthenticatedIdentities: true }).attrId;
+const identityPoolId = 'us-west-2:54ee61c9-8d51-46aa-8898-33e8578d2f75'; // Random id, value not used in this test
+const appsync = new ServicePrincipal('appsync.amazonaws.com');
+const authenticatedUserRole = new Role(stack, 'TestAuthRole', { assumedBy: appsync });
+const unauthenticatedUserRole = new Role(stack, 'TestUnauthRole', { assumedBy: appsync });
+const functionAuthorizer = new NodejsFunction(stack, 'TestAuthorizer', { entry: path.join(__dirname, 'authorizer.ts') });
+const oneHour = Duration.hours(1);
+const oneWeek = Duration.days(7);
+const oidcName = 'my-oidc-provider';
+const oidcUrl = 'https://oidc.example.com';
+
+new AmplifyGraphqlApi(stack, 'TestAuthRules', {
+  definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+    type Todo
+      @model
+      @auth(
+        rules: [
+          { allow: public, provider: apiKey }
+          { allow: public, provider: iam }
+          { allow: owner, ownerField: "userPoolOwner", provider: userPools }
+          { allow: owner, ownerField: "oidcOwner", provider: oidc }
+          { allow: private, provider: userPools }
+          { allow: private, provider: oidc }
+          { allow: private, provider: iam }
+          { allow: groups, groups: ["Admin"], provider: userPools }
+          { allow: groups, groups: ["Admin"], provider: oidc }
+          { allow: custom, provider: function }
+        ]
+      ) {
+      id: ID!
+    }
+  `),
+  authorizationModes: {
+    defaultAuthorizationMode: 'AWS_IAM',
```

Comment:
Do we expect any behavior difference if IAM is an additional mode rather than the default?

Reply:
No, this is just a prop which determines the order in which we specify the `default` mode vs. `additional` modes in the underlying L1 construct.